src/garage/envs/base.py

```diff
@@ -140,15 +141,29 @@ def step(self, action):
         """
         return self.env.step(action)
 
-    def __setstate__(self, state):
```

Comment:
In `__getstate__`, we didn't remove the viewer object for `self` (original env object), rather we make a copy and return that copy as the state. After unpickling, the viewer object in the newly constructed env is gone, there is no way we can populate it back unless we construct a new one (because we remove that in the state). However, the viewer in `self` is still there.

Reply:
@nish21 Does that make sense?

Reply:
But could you verify the behavior with current __setstate__? I think the current implementation of setstate restores the viewer as it's creating a new object

Reply:
I think what I did was, while pickling, I don't pickle the viewer object -- remove viewer from the copy of dict. But during setstate, i restore the viewer and updating the params with the pickled object params.

Reply:
How did you restore the viewer object? Where is the viewer object kept and how is it passed to the newly unpickled env?

Reply:
to be clear, we probably don't want to restore viewer state on environments while unpickling. The most common pickling case for an environment is to transfer it from the main training to a worker process. We don't want a worker process spending cycles on rendering.